iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/hsm/HttpsHsmClient.java

```diff
@@ -251,17 +251,23 @@ private HttpsResponse sendHttpRequestUsingUnixSocket(HttpsRequest httpsRequest,
 
             if (httpsRequest.getBody() != null)
             {
-                //append http request body to the request bytes
-                ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-                outputStream.write(requestBytes);
-                outputStream.write(httpsRequest.getBody());
-
-                log.trace("Writing {} bytes to unix domain socket", outputStream.size());
-                unixDomainSocketChannel.write(outputStream.toByteArray());
+                // closes the output stream when it exits this block
+                try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream())
+                {
+                    //append http request body to the request bytes
+                    outputStream.write(requestBytes);
+                    outputStream.write(httpsRequest.getBody());
+
+                    byte[] output = outputStream.toByteArray();
+                    log.trace("Writing {} bytes to unix domain socket", output.length);
+                    log.trace("{}", output);
```

Comment:
Maybe specify what we are logging here?
```suggestion
                    log.trace("Contents of the request:\r\n{}", output);
```

Reply:
Sure thing